homeassistant/loader.py

```diff
@@ -851,17 +855,35 @@ async def async_get_component(self) -> ComponentProtocol:
             self.pkg_path not in sys.modules
             or (self.config_flow and f"{self.pkg_path}.config_flow" not in sys.modules)
         )
-        if load_executor:
-            try:
-                comp = await self.hass.async_add_import_executor_job(self.get_component)
-            except ImportError as ex:
-                load_executor = False
-                _LOGGER.debug("Failed to import %s in executor", domain, exc_info=ex)
-                # If importing in the executor deadlocks because there is a circular
-                # dependency, we fall back to the event loop.
+        self._component_future = self.hass.loop.create_future()
```

Comment:
Should we limit this to `if load_executor` ? Otherwise there is no `await` and it cannot ever be called twice

Reply:
I think so

Reply:
5181e24c8fcfc8e05f96e0068aba15f918494059